storage/s3/src/test/java/io/aiven/kafka/tieredstorage/storage/s3/S3MultiPartOutputStreamTest.java

```diff
@@ -282,4 +320,10 @@ private static void assertAbortMultipartUploadRequest(final AbortMultipartUpload
         assertThat(request.getKey()).isEqualTo(FILE_KEY);
         assertThat(request.getUploadId()).isEqualTo(UPLOAD_ID);
     }
+
+    private void assertStreamIsClosed(final S3MultiPartOutputStream out) {
+        assertThat(out.isClosed()).isTrue();
+        // retry close to validate no exception is thrown and number of calls to complete/upload does not change
```

Comment:
Where do we actually verify the number of calls?

Reply:
it's on each test case as it depends on the test condition. 
i guess would it be better to pull those in here? can give a try.

Reply:
actually, taking this out to the methods itself. A bit of repeated code, but clearer to assess within each test case, wdyt?

Reply:
I agree but the comment is still misleading. After you added an assertions it says on it's own that we are testing that `close` does not throw exception. For the number of calls we are sometimes(not everywhere) are checking the expected number of calls for the whole test, not that the number of calls does not change(e.g. verifyNoMoreInterations or similar). So I would drop those comments altogether.